core/src/main/java/org/elasticsearch/index/shard/ShadowIndexShard.java

```diff
@@ -102,4 +103,22 @@ public boolean shouldFlush() {
     public TranslogStats translogStats() {
         return null; // shadow engine has no translog
     }
+
+
+    @Override
+    public void updateGlobalCheckpointOnReplica(long checkpoint) {
+        // nocommit: think shadow replicas through
```

Comment:
yeah. I want to take some time to think about the whole push in the light of seq_no, but I'm planning to let "normal" shard work to stabilize first.